### PR TITLE
Allow source replacement in AudioRecorder to prevent the need to recreate Service when source changes.

### PR DESCRIPTION
**Pull request by lduffy-voysis**
- added Source interface for file and recorder source types. Replaced wavAudioRecorder with fileAudioSource

- moved recorder to constructor in wakeWordDetector. Removed recorder from wakewordServiceImpl

### CODE REVIEW

File: library/src/main/java/com/voysis/recorder/FileAudioSource.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.voysis.recorder
+
+import com.voysis.recorder.AudioRecorderImpl.Companion.DEFAULT_READ_BUFFER_SIZE
+import java.io.File
+import java.io.InputStream
+
+class FileAudioSource(var wavFile: File? = null) : Source {
+
+    private var inputStream: InputStream? = null
+    private var bytesRead = 1
```
Comment (jcftang) [inline library/src/main/java/com/voysis/recorder/FileAudioSource.kt]: is this meant to be a counter on the number of bytes read from the file? why is this initialised to 1 and not 0?

Comment (lduffy-voysis) [reply library/src/main/java/com/voysis/recorder/FileAudioSource.kt]: yeah that should initially be 0. updated.

File: library/src/main/java/com/voysis/recorder/FileAudioSource.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    override fun startRecording() {
+        inputStream = wavFile!!.inputStream()
+    }
+
+    override fun generateMimeType(): MimeType? = MimeType(16000, 16, "signed-int", false, 1)
+
+    override fun generateBuffer(): ByteArray = ByteArray(DEFAULT_READ_BUFFER_SIZE)
+
+    override fun isRecording(): Boolean {
+        return bytesRead > -1
```
Comment (jcftang) [inline library/src/main/java/com/voysis/recorder/FileAudioSource.kt]: at which part of the implementation does bytesRead gets set to -1?

Comment (lduffy-voysis) [reply library/src/main/java/com/voysis/recorder/FileAudioSource.kt]: when there are no more bytes to be read from the inputStream it returns -1

Comment (jcftang) [reply library/src/main/java/com/voysis/recorder/FileAudioSource.kt]: ok, I missed that.

**Pull request merged**